collection-scripts/gather_host_network_logs

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+source $(dirname "$0")/common.sh
+get_log_collection_args
+. gather_service_logs_util
+
+BASE_COLLECTION_PATH="/must-gather"
+HOST_NETWORK_NODE_LOGS=$BASE_COLLECTION_PATH/host_network_logs
+
+mkdir -p "${HOST_NETWORK_NODE_LOGS}"/
+
+PIDS=()
+echo "INFO: Collecting host networking logs"
+
+collect_service_logs --role=master "on-prem-resolv-prepender"
+collect_service_logs --role=worker "on-prem-resolv-prepender"
+
+collect_service_logs --role=master "nodeip-configuration"
+collect_service_logs --role=worker "nodeip-configuration"
+
+CLUSTER_NODES="${@:-$(oc get node -l node-role.kubernetes.io/master -oname)}"
```

Comment:
Gathering logs from the workers (when there are 500 nodes) might be a problem is there any way to limit this ?

Reply:
Can we make it so that this is something a user pass in (the list of nodes to collect this data from)?

Reply:
I feel like this would defeat the purpose. Maybe I never defined the problem we have correctly - what happens today is that people with "some networking problem" (as they feel) come to our team with the most basic must-gather and ask for help. At this moment we start playing ping-pong, because our first reply is "please bring sosreport from all the nodes". Only based on sosreport we can say if the problem is really on us, or not at all. If yes, we ask for the next set of logs (networkmanager with `level=TRACE`).

With this PR I aimed to at least bring must-gather and sosreport a bit closer, so that customer coming with must-gather collected using this PR would receive as a next instruction "yes, it's networking, please bring sosreport and NM trace logs" or "no, it's not networking" already.

If now we make it configurable, I do not believe any customer will ever come with a correct must-gather in the first place (as in, they will first give us must-gather without logs from any worker what renders useless).

Reply:
I think we need to time box this (you only get N (<5) minutes to collect all this (or this script exits), or find some way to potentially limit or target this collection at a list of nodes.  

IMO; the exposure risk of trying to collect all of this on hundreds of nodes; add too much to the collection and/or takes too much time (so we have to address that).

Reply:
As a trade-off I am okay to limit this only to master nodes, is this acceptable for you? It would still put us in a better situation than we are today and if after some months we still feel that what support&customers bring us does not make sense, we can reiterate on this discussion